src/client/datascience/raw-kernel/rawJupyterSession.ts

```diff
@@ -109,24 +114,51 @@ export class RawJupyterSession extends BaseJupyterSession {
         } catch (error) {
             this.connected = false;
             if (error instanceof CancellationError) {
-                sendTelemetryEvent(Telemetry.RawKernelSessionStartUserCancel);
+                sendKernelTelemetryEvent(resource, Telemetry.RawKernelSessionStart, stopWatch.elapsedTime, {
```

Comment:
Use a new telemetry event.
This way we can use that single event & check the property for:
* success or failed
* reason if failed
With the queries we can easily group these, etc (i.e single event with props is more powerful for analysis).

Else if things fail, we have to use multiple events today to figure out why it failed & the like.

Reply:
Eg. if we wanted to take a count of success vs failure, then we have to do a join, this way we can just take a count grouped by `reason`.

Reply:
Is this same sort of detail sent for jupyter kernel failures? I didn't think it was but maybe I'm missing something.